### ISSUE TITLE
Conflict between "outscale_vm" and "outscale_nic" when creating a VM with multiple nics

### ISSUE THREAD

**Issue by outscale-toa** [outscale/terraform-provider-outscale#50]
### Terraform Version

```
Terraform v1.1.3
on linux_amd64
+ provider registry.terraform.io/outscale-dev/outscale v0.5.1
```

### Terraform Configuration Files

```
resource "outscale_net" "outscale_net8" {
    ip_range = "10.0.0.0/16"
}
resource "outscale_subnet" "outscale_subnet8" {
 net_id              = outscale_net.outscale_net8.net_id
 ip_range            = "10.0.0.0/24"
 subregion_name      = "${var.region}a"
}
resource "outscale_security_group" "outscale_security_group8" {
    description         = "test vm with nic"
    security_group_name = "private-sg-1"
    net_id              = outscale_net.outscale_net8.net_id
}

resource "outscale_security_group" "outscale_security_group10" {
    description         = "test vm with nic-10"
    security_group_name = "private-sg-10"
    net_id              = outscale_net.outscale_net8.net_id
}

resource "outscale_nic" "outscale_nic8" {
    subnet_id = outscale_subnet.outscale_subnet8.subnet_id
    security_group_ids = [outscale_security_group.outscale_security_group8.security_group_id]
}

resource "outscale_nic" "outscale_nic9" {
    subnet_id = outscale_subnet.outscale_subnet8.subnet_id
    security_group_ids = [outscale_security_group.outscale_security_group10.security_group_id]
}
resource "outscale_vm" "outscale_vm8" {
    image_id            = var.image_id
    vm_type             = "tinav4.c2r2p2"
    keypair_name        = var.keypair_name
    nics       {
       nic_id =  outscale_nic.outscale_nic8.nic_id
       device_number = "0"
     }
}

resource "outscale_nic_link" "link-1" {
    device_number = 1
    vm_id = outscale_vm.outscale_vm8.vm_id
    nic_id = outscale_nic.outscale_nic9.nic_id
}
```

### Debug Output

When applying the configuration twice without any change, terraform plan to destroy the vm and recreate it. 
```
$ terraform apply
outscale_net.outscale_net8: Refreshing state... [id=vpc-b4720494]
outscale_subnet.outscale_subnet8: Refreshing state... [id=subnet-b0fd158e]
outscale_security_group.outscale_security_group8: Refreshing state... [id=sg-a3ebedb3]
outscale_security_group.outscale_security_group10: Refreshing state... [id=sg-c1d2508d]
outscale_security_group.outscale_security_group9: Refreshing state... [id=sg-75b90bee]
outscale_nic.outscale_nic8: Refreshing state... [id=eni-bb477bfe]
outscale_nic.outscale_nic9: Refreshing state... [id=eni-47eaac34]
outscale_vm.outscale_vm8: Refreshing state... [id=i-17d606a1]
outscale_nic_link.link-1: Refreshing state... [id=eni-attach-7e77a892]

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
-/+ destroy and then create replacement

Terraform will perform the following actions:

  # outscale_nic_link.link-1 must be replaced
-/+ resource "outscale_nic_link" "link-1" {
      ~ delete_on_vm_deletion = false -> (known after apply)
        device_number         = 1
      ~ id                    = "eni-attach-7e77a892" -> (known after apply)
      ~ link_nic_id           = "eni-attach-7e77a892" -> (known after apply)
        nic_id                = "eni-47eaac34"
      ~ request_id            = "a402f452-2d7e-4172-84a3-22fe4b41f03e" -> (known after apply)
      + state                 = (known after apply)
      + vm_account_id         = (known after apply)
      ~ vm_id                 = "i-17d606a1" -> (known after apply) # forces replacement
    }

  # outscale_vm.outscale_vm8 must be replaced
-/+ resource "outscale_vm" "outscale_vm8" {
      + admin_password                 = (known after apply)
      ~ architecture                   = "x86_64" -> (known after apply)
      ~ bsu_optimized                  = false -> (known after apply)
      + client_token                   = (known after apply)
      ~ deletion_protection            = false -> (known after apply)
      ~ hypervisor                     = "xen" -> (known after apply)
      ~ id                             = "i-17d606a1" -> (known after apply)
        image_id                       = "ami-5c8b4adf"
      ~ is_source_dest_checked         = true -> (known after apply)
        keypair_name                   = "New-IN1-KP"
      ~ launch_number                  = 0 -> (known after apply)
      ~ net_id                         = "vpc-b4720494" -> (known after apply)
      + os_family                      = (known after apply)
      ~ performance                    = "high" -> (known after apply)
      ~ placement_subregion_name       = "in-west-1a" -> (known after apply)
      ~ placement_tenancy              = "default" -> (known after apply)
      ~ private_dns_name               = "ip-10-0-0-61.in-west-1.compute.internal" -> (known after apply)
      ~ private_ip                     = "10.0.0.61" -> (known after apply)
      ~ product_codes                  = [
          - "0001",
        ] -> (known after apply)
      + public_dns_name                = (known after apply)
      + public_ip                      = (known after apply)
      ~ request_id                     = "67df322e-f9e6-4db5-8e8e-fb7c467c6716" -> (known after apply)
      ~ reservation_id                 = "r-8986f943" -> (known after apply)
      ~ root_device_name               = "/dev/sda1" -> (known after apply)
      ~ root_device_type               = "ebs" -> (known after apply)
      ~ security_groups                = [
          - {
              - security_group_id   = "sg-a3ebedb3"
              - security_group_name = "private-sg-1"
            },
        ] -> (known after apply)
      ~ state                          = "running" -> (known after apply)
      + state_reason                   = (known after apply)
      ~ subnet_id                      = "subnet-b0fd158e" -> (known after apply)
      + user_data                      = (known after apply)
      ~ vm_id                          = "i-17d606a1" -> (known after apply)
      ~ vm_initiated_shutdown_behavior = "stop" -> (known after apply)
        vm_type                        = "tinav4.c2r2p2"

      ~ block_device_mappings_created {
          ~ bsu         = {
              - "delete_on_vm_deletion" = "true"
              - "link_date"             = "2021-06-08T16:17:55.439Z"
              - "state"                 = "attached"
              - "volume_id"             = "vol-ff833812"
            } -> (known after apply)
          ~ device_name = "/dev/sda1" -> (known after apply)
        }

      - nics { # forces replacement
          - account_id                 = "108845946057" -> null
          - delete_on_vm_deletion      = false -> null
          - device_number              = 0 -> null
          - is_source_dest_checked     = true -> null
          - link_nic                   = [
              - {
                  - delete_on_vm_deletion = false
                  - device_number         = "0"
                  - link_nic_id           = "eni-attach-2fbea812"
                  - state                 = "attached"
                },
            ] -> null
          - link_public_ip             = [] -> null
          - mac_address                = "aa:a3:a6:ce:05:21" -> null
          - net_id                     = "vpc-b4720494" -> null
          - nic_id                     = "eni-bb477bfe" -> null
          - private_dns_name           = "ip-10-0-0-61.in-west-1.compute.internal" -> null
          - secondary_private_ip_count = 0 -> null
          - security_group_ids         = [
              - "sg-a3ebedb3",
            ] -> null
          - security_groups            = [
              - {
                  - security_group_id   = "sg-a3ebedb3"
                  - security_group_name = "private-sg-1"
                },
            ] -> null
          - state                      = "in-use" -> null
          - subnet_id                  = "subnet-b0fd158e" -> null

          - private_ips {
              - is_primary       = true -> null
              - link_public_ip   = [] -> null
              - private_dns_name = "ip-10-0-0-61.in-west-1.compute.internal" -> null
              - private_ip       = "10.0.0.61" -> null
            }
        }
      - nics { # forces replacement
          - account_id                 = "108845946057" -> null
          - delete_on_vm_deletion      = false -> null
          - device_number              = 1 -> null
          - is_source_dest_checked     = true -> null
          - link_nic                   = [
              - {
                  - delete_on_vm_deletion = false
                  - device_number         = "1"
                  - link_nic_id           = "eni-attach-7e77a892"
                  - state                 = "attached"
                },
            ] -> null
          - link_public_ip             = [] -> null
          - mac_address                = "aa:cc:70:98:34:3b" -> null
          - net_id                     = "vpc-b4720494" -> null
          - nic_id                     = "eni-47eaac34" -> null
          - private_dns_name           = "ip-10-0-0-233.in-west-1.compute.internal" -> null
          - secondary_private_ip_count = 0 -> null
          - security_group_ids         = [
              - "sg-c1d2508d",
            ] -> null
          - security_groups            = [
              - {
                  - security_group_id   = "sg-c1d2508d"
                  - security_group_name = "private-sg-10"
                },
            ] -> null
          - state                      = "in-use" -> null
          - subnet_id                  = "subnet-b0fd158e" -> null

          - private_ips {
              - is_primary       = true -> null
              - link_public_ip   = [] -> null
              - private_dns_name = "ip-10-0-0-233.in-west-1.compute.internal" -> null
              - private_ip       = "10.0.0.233" -> null
            }
        }
      + nics { # forces replacement
          + account_id                 = (known after apply)
          + delete_on_vm_deletion      = (known after apply)
          + description                = (known after apply)
          + device_number              = 0
          + is_source_dest_checked     = (known after apply)
          + link_nic                   = (known after apply)
          + link_public_ip             = (known after apply)
          + mac_address                = (known after apply)
          + net_id                     = (known after apply)
          + nic_id                     = "eni-bb477bfe"
          + private_dns_name           = (known after apply)
          + secondary_private_ip_count = (known after apply)
          + security_group_ids         = []
          + security_groups            = (known after apply)
          + state                      = (known after apply)
          + subnet_id                  = (known after apply)

          + private_ips {
              + is_primary       = (known after apply)
              + link_public_ip   = (known after apply)
              + private_dns_name = (known after apply)
              + private_ip       = (known after apply)
            }
        }
    }

Plan: 2 to add, 0 to change, 2 to destroy.

```
### Crash Output

N/A
### Expected Behavior

When The same configuration is applied twice, no change is applied.

### Actual Behavior

Terraform plan to destroy "outscale_vm" and "outscale_nic_link" resources and recreate them.

### Steps to Reproduce

Apply the configuration file described above twice in a row.
1. `terraform init`
2. `terraform apply`
3. ` terraform apply`

**Comment by boussoufiane** [comment]
Hello , 

@outscale-toa  Do you have an idea when this issue bug will be fixed ?
Is there a workaround to avoid this issue ?

Best regards,
Yassine

**Comment by outscale-toa** [comment]
Hello @boussoufiane 

We plan to fix it in next release (v0.6.0)

Thanks,

**Comment by outscale-rce** [comment]
This behavior is due to the "nics" parameters within the VM block. With this configuration, the issue does not occur.

```
resource "outscale_net" "net02" {
    ip_range = "10.10.0.0/16"
    tags {
        key   = "name"
        value = "terraform-net-for-vm-with-nic"
    }
}

resource "outscale_subnet" "subnet02" {
    net_id         = outscale_net.net02.net_id
    ip_range       = "10.10.0.0/24"
    subregion_name = "eu-west-2a"
    tags {
        key   = "name"
        value = "terraform-subnet-for-vm-with-nic"
    }
}

resource "outscale_security_group" "security_group01" {
  description         = "vm security group"
  security_group_name = "vm_security_group1"
  net_id         = outscale_net.net02.net_id
}

resource "outscale_nic" "nic01" {
    subnet_id = outscale_subnet.subnet02.subnet_id
    security_group_ids  = [outscale_security_group.security_group01.security_group_id]
}

resource "outscale_nic" "nic02" {
    subnet_id = outscale_subnet.subnet02.subnet_id
    security_group_ids  = [outscale_security_group.security_group01.security_group_id]
}

resource "outscale_vm" "vm01" {
    image_id             = var.image_id
    vm_type              = var.vm_type
    keypair_name         = outscale_keypair.my_keypair.keypair_name
    primary_nic {
        nic_id        = outscale_nic.nic01.nic_id
        device_number = "0"
    }
}

resource "outscale_nic_link" "nic_link02" {
    device_number = "1"
    vm_id         = outscale_vm.vm01.vm_id
    nic_id        = outscale_nic.nic02.nic_id
}
```
